c10/test/util/AutoTimer_test.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Ported from folly/logging/test/AutoTimerTest.cpp
+ */
+
+#include <c10/util/AutoTimer.h>
+#include <c10/util/ScopeExit.h>
+#include <gtest/gtest.h>
+
+using namespace c10;
+
+struct StubLogger {
+  void operator()(std::string_view msg, std::chrono::duration<double> sec) {
+    m = std::string(msg);
+    t = sec.count();
+  }
+  static std::string m;
+  static double t;
+};
+
+std::string StubLogger::m = "";
+double StubLogger::t = 0;
```

Comment:
nit: since this is just a test you can definitely take advantage of C++17 `inline static`, search https://en.cppreference.com/w/cpp/language/static for "A static data member may be declared inline"